### ISSUE TITLE
Bug, or am I misinterpreting? required options used out of order get wrong option argument when long and short options are possible

### ISSUE THREAD

**Issue by alaricljs** [docopt/docopt#258]
This also happens on the web demo, so [here it is] (http://try.docopt.org/?doc=Usage%3A%0D%0A++x++++++%28-a+%3Ca%3E+|+--able+%3Ca%3E%29+%28-b+%3Cb%3E+|+--bee+%3Cb%3E%29%0D%0A++x+-h+|+--help%0D%0A++x+-v+|+--version%0D%0A%0D%0AOptions%3A%0D%0A++-a|--able+%3Ca%3E++++aaaaaa%0D%0A++-b|--bee+%3Cb%3E+++++bbbbb%0D%0A++-h+--help++++++++show+this+screen%0D%0A++-v+--version+++++version%0D%0A&argv=-b+b+-a+a)

Basically, having long and short options w/ an argument causes the arguments to be picked up in the order specified in the Usage instead of the order specified in the actual command line itself.
Usage:  x (-a <a> | --able <a>) (-b <b> | -bee <b>)
x -b b -a a
<a> = b  (wrong, should be a)
<b> = a  (wrong, should be b)

This works 
Usage: x -a <a> -b <b>
x -b b -a a
<a> = a
<b> = b

**Comment by BurntSushi** [comment]
As per the spec:

> It is necessary to list option descriptions in order to specify:
>
> * synonymous short and long options,
> * if an option has an argument,
> * if option's argument has a default value.

So you need to specify that the flags have arguments:

```
Usage: x (-a <a> | --able <a>) (-b <b> | --bee <b>)

Options:
  -a, --able <a>
  -b, --bee <b>
```

**Comment by alaricljs** [comment]
Sorry, that was an incomplete representation, the whole thing was linked to at the very start.  Apparently my issue is that I didn't get this from the docs:
Options:
  -a, --able \<a\>     IS NOT the same as -a|--able \<a\>

**Comment by sfriesel** [comment]
afaict, this can be closed
